### ISSUE TITLE
reading configuration file is broken for a long time

### ISSUE THREAD

**Issue by yilmazdurmaz** [htmlhint/HTMLHint#504]
**Describe the bug**
seemingly working function, reading configuration from a file, is in fact broken. it simply does not not work.

**To Reproduce**
- create a new html file, and fill base content with emmet (i use vscode).
- now remove the title tag, and make a typo to closing body tag "bodx". 3 errors must be present.
- create a config file and add {"title-require": true}. now only title error shows and tag match errors are gone.
- use {"title-require": false} and no errors are present.
- use {"title-require": false,} (trailing comma) and all errors are back
- use arbitrary {"a":"a"} and {"a":"a",} (trailing comma) and see that previous 2 steps has nothing to do with what used inside

**Expected behavior**
- a single rule change should not need trailing comma
- a single rule change should not affect other default rules
- an arbitrary input should not overwrite any default rule.

**Screenshots**
- no screenshot needed, steps are clear

**Desktop (please complete the following information):**

- OS: windows
- Version 10 1909

**Additional context**
from vscode
  - vscode:  1.41.1 and 1.47.1
  - HTMLHint extension v0.8.0 (htmlhint 0.11.0)

from terminal
 - Node: 12.14.1
 - htmlhint: 0.14.1

**Comment by Shinigami92** [comment]
Tested it with your changes in branch `feat/add-config-attribute-extends` and it seems to be fixed
So I think this can be closed when we released the `1.0.0`

**Comment by yilmazdurmaz** [comment]
"It will be closed if no further activity occurs"

closing a bug topic without a solution? is this some setting missing in the project's github settings or a behaviour of github in general?

**Comment by yilmazdurmaz** [comment]
@Shinigami92 
is there an estimate when v1.0 is coming? or anything to integrate the fix into current v0.x line?

**Comment by yilmazdurmaz** [comment]
is this problem solved in v0.14.2? or still delayed for v1.0.0

**Comment by polyzen** [comment]
This is still an issue, please tag with `keep-unstale`.

**Comment by polyzen** [comment]
@thedaviddias, please re-open as well. :)

**Comment by KhafraDev** [comment]
Just had this issue and found a workaround. When supplying your own valid config, it seems to override the default values, meaning that all rules get disabled. To fix this, create a `.htmlhintrc` file and fill it in as follows:

```json
{
  "tagname-lowercase": true,
  "attr-lowercase": true,
  "attr-value-double-quotes": true,
  "doctype-first": true,
  "tag-pair": true,
  "spec-char-escape": true,
  "id-unique": true,
  "src-not-empty": true,
  "attr-no-duplication": true,
  "title-require": true
}

```

After this you can modify/add/delete rules as you please.

**Comment by coliff** [comment]
yep, think this is fixed now - please comment if still an issue.

**Comment by yilmazdurmaz** [comment]
@coliff , unfortunately the problem is still there.

I installed 1.3.0 and used it on a freshly emmet-filled HTML file. It complains from single quotes in 5 lines.

I created a ".htmlhintrc" file and used arbitrary `{"a":a}` and it finely said "Config could not be parsed".

But using a meaningful tag setting it still behaves the same:
- using `{"title-require": true}` overwrites all defaults to void and only uses rules in this file.
- `{"title-require": true,}` (trailing comma) gives "Config could not be parsed" error

**Comment by yilmazdurmaz** [comment]
> Config could not be parsed

I was not in constant development, and had been away from it for a long time. 

At the time, I was using VSCode and opened this issue according to my observations, so I might be wrong on my tests that:
- errors were shown because init file was not parsed, so default were in use
- errors were suppressed because init file voids all defaults.

In that case, the trailing comma and arbitrary values were causing parse errors, and that is normal.

But, shouldn't the defaults be kept when reading from the init file? Shouldn't they be combined instead?

**Comment by coliff** [comment]
If you use your own config then it doesn't use the defaults at all... they don't combine and that's by design. If you're starting a new project then I recommend adding a config like this and enabling the rules that you want:

https://htmlhint.com/configuration/#example-configuration-file
